token/program-2022/src/extension/confidential_mint_burn/processor.rs

```diff
@@ -126,10 +127,11 @@ fn process_update_decryptable_supply(
     check_program_account(mint_info.owner)?;
     let mint_data = &mut mint_info.data.borrow_mut();
     let mut mint = PodStateWithExtensionsMut::<PodMint>::unpack(mint_data)?;
+    let mint_authority = mint.base.mint_authority;
     let mint_burn_extension = mint.get_extension_mut::<ConfidentialMintBurn>()?;
 
-    let authority = Option::<Pubkey>::from(mint_burn_extension.authority)
-        .ok_or(TokenError::NoAuthorityExists)?;
+    let authority = OptionalNonZeroPubkey::try_from(mint_authority)?;
+    let authority = Option::<Pubkey>::from(authority).ok_or(TokenError::NoAuthorityExists)?;
```

Comment:
Same here, can use `ok_or`:
```suggestion
    let authority = mint.base.mint_authority.ok_or(TokenError::NoAuthorityExists)?;
```